js_modules/dagster-ui/packages/ui-core/src/runs/CapturedLogPanel.tsx

```diff
@@ -28,34 +30,98 @@ interface CapturedOrExternalLogPanelProps extends CapturedLogProps {
 
 export const CapturedOrExternalLogPanel = React.memo(
   ({logCaptureInfo, ...props}: CapturedOrExternalLogPanelProps) => {
+    const getShellCmd = (ioType: string, logCaptureInfo: ILogCaptureInfo | undefined) => {
+      if (logCaptureInfo?.logManagerMetadata) {
+        const path =
+          ioType === 'stdout' ? logCaptureInfo.stdoutUriOrPath : logCaptureInfo.stderrUriOrPath;
+
+        try {
+          const metadata = JSON.parse(logCaptureInfo.logManagerMetadata);
+          switch (metadata.type) {
+            case 'AzureBlobComputeLogManager':
+              if (metadata.storage_account && metadata.container) {
+                return `az storage blob download --account-name ${metadata.storage_account} --container-name ${metadata.container} --name ${path}`;
+              }
+          }
+        } catch {
+          return undefined;
+        }
+      }
+      return undefined;
+    };
+
     const externalUrl =
       logCaptureInfo &&
       (props.visibleIOType === 'stdout'
         ? logCaptureInfo.externalStdoutUrl
         : logCaptureInfo.externalStderrUrl);
-    if (externalUrl) {
+    const uriOrPath =
+      logCaptureInfo &&
+      (props.visibleIOType === 'stdout'
+        ? logCaptureInfo.stdoutUriOrPath
+        : logCaptureInfo.stderrUriOrPath);
+    const shellCmd = getShellCmd(props.visibleIOType, logCaptureInfo);
+
+    const copy = useCopyToClipboard();
+    const onClickFn = async (key: string, value: string | undefined) => {
+      if (!value) {
```

Comment:
the check is necessary because value could be undefined.

Reply:
what does it show if the value is undefined?

Reply:
actually it's more dealing with the type checker here.
the function cannot be called because the blocks that would make it available would not be rendered